src/features/hooks/use-signin-and-signup/index.tsx

```diff
@@ -0,0 +1,119 @@
+import { useEffect, useState } from 'react'
+import apiManager from 'common/websocket'
+import { isBrowser } from 'common/utility'
+
+export const useSigninAndSignup = (platform: string, token: string) => {
+    const [loading, setLoading] = useState(false)
+    const [has_account, setHasAccount] = useState(null)
+    const [authorized, setAuthorized] = useState(false)
+    const [account_error, setAccountError] = useState(null)
+    const [create_account_error, setCreateAccountError] = useState(null)
+    const [service_token, setServiceToken] = useState(null)
+
+    useEffect(() => {
+        if (token) {
+            setLoading(true)
+            apiManager
+                .augmentedSend('authorize', {
+                    authorize: token,
+                })
+                .then(() => {
+                    setAuthorized(true)
+                })
+                .catch((reason) => {
+                    setAccountError(reason.error.message)
+                    setLoading(false)
+                })
+        } else {
+            setAccountError('Token in URL is required as token1=')
+        }
+    }, [token])
+
+    useEffect(() => {
+        if (authorized) {
+            apiManager
+                .augmentedSend('trading_platform_accounts', {
+                    trading_platform_accounts: 1,
+                    platform,
+                })
+                .then((response) => {
+                    if (response.trading_platform_accounts.length > 0) {
+                        setHasAccount(true)
+                    } else {
+                        setHasAccount(false)
+                    }
+                })
+                .catch((reason) => {
+                    setAccountError(reason.error.message)
+                    setLoading(false)
+                })
+        }
+    }, [authorized, platform])
+
+    useEffect(() => {
```

Comment:
Thanks @mohsen-deriv for comments. I've fixed 2 of them. but for this one:
`Layout and SEO handling should happen in page level. please check out the src/features/home as an example.`
Which component are you referring we should apply this change?

Reply:
you can forget about the third point man, I see now why you did it like that

Reply:
Since all these cTrader pages are excluded from sitemap, We didn't use `SEO` component in these pages.
But about the `Layout` is there something that I've missed?